### PR TITLE
Add wlr_fractional_scale_v1

### PR DESCRIPTION
**Pull request by MaxVerevkin**

### CODE REVIEW

**Comment by MaxVerevkin** [comment]
Tested with https://github.com/MaxVerevkin/river/tree/fractional-scaling

File: src/types/fractional_scale_v1.zig [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const wlr = @import("../wlroots.zig");
+
+const wayland = @import("wayland");
+const wl = wayland.server.wl;
+
+pub const FractionalScaleManagerV1 = extern struct {
+    global: *wl.Global,
+
+    events: extern struct {
+        destroy: wl.Signal(void),
```
Comment (novakne) [inline src/types/fractional_scale_v1.zig:10]:
```suggestion
        destroy: wl.Signal(*FractionalScaleManagerV1),
```
I think that should be that here

Comment (MaxVerevkin) [reply src/types/fractional_scale_v1.zig:10]: This is a copy-paste from `viewporter.zig`, is it wrong too?

Comment (ifreund) [reply src/types/fractional_scale_v1.zig:10]: Copy pasting is always wrong, the wlroots source code must be examined to determine what pointer if any is passed as the data argument when signals are emitted.

Comment (MaxVerevkin) [reply src/types/fractional_scale_v1.zig:10]: Okay, wlroots passes NULL. void is good then?

Comment (ifreund) [reply src/types/fractional_scale_v1.zig:10]: Yep, void is correct

**Review by ifreund — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**